### PR TITLE
Check as-cran + default landing page

### PR DESCRIPTION
**Pull request by arkadiuszbeer**
Solves these two issues:
https://github.com/insightsengineering/idr-tasks/issues/477
https://github.com/insightsengineering/idr-tasks/issues/478

### CODE REVIEW

**Comment by Melkiades** [comment]
Noice!!! 👍

File: .github/workflows/check.yaml [hunk 1]
```diff
@@ -24,6 +24,12 @@ jobs:
     uses: insightsengineering/r.pkg.template/.github/workflows/build-check-install.yaml@main
     secrets:
       REPO_GITHUB_TOKEN: ${{ secrets.REPO_GITHUB_TOKEN }}
+    with:
+      additional-env-vars: |
+        _R_CHECK_CRAN_INCOMING_=false
+        _R_CHECK_CRAN_INCOMING_REMOTE_=false
```
Comment (Melkiades) [inline .github/workflows/check.yaml:30]: why incoming is false?

Comment (arkadiuszbeer) [reply .github/workflows/check.yaml:30]: We dont want to have errors like "Strong dependencies not in mainstream repositories"
https://github.com/insightsengineering/teal.modules.clinical/actions/runs/4142734635/jobs/7163754425#step:27:60
Only _R_CHECK_CRAN_INCOMING_REMOTE_ is needed.

**Comment by pawelru** [comment]
Let me jump into this PR as `tern` is the closest to CRAN submission. 
My long term goal for this exercise is to find collection of env vars so that R CMD CHECK has zero notes and then we can fail our check-GHA on error, warning **and** notes. Currently there are some warning/notes that needs to be fixed so here in this PR I would like to achieve the state of leaving only _fixable_ notes. 

At the current state we have the following:
```
* checking CRAN incoming feasibility ... NOTE
Maintainer: ‘Joe Zhu <joe.zhu@roche.com>’

Version contains large components (0.7.10.9042)
```
We can't avoid that until we release to CRAN. For some of the packages that would take some time. Therefore I propose to silent this note.
This consists of two parts - large version components as well as new maintainer. I looked into the [source code of R](https://github.com/wch/r-source/blob/9065779ee510b7bd8ca93d08f4dd4b6e2bd31923/src/library/tools/R/QC.R#L7179-L7183) and I have found that there might be a env var that could silent it - `_R_CHECK_CRAN_INCOMING_SKIP_LARGE_VERSION_`. However, I cannot find any(!) documentation about this so I might be wrong. For the latter part - there is a `_R_CHECK_CRAN_INCOMING_` env var. It might happen that setting this one would also disable large version check so I would suggest to check this one first.

```
* checking installed package size ... NOTE
  installed size is  6.5Mb
  sub-directories of 1Mb or more:
    doc   4.5Mb
```
This note is acceptable thus I propose to silent it. There are two env vars for it: [`_R_CHECK_PKG_SIZES_`](https://rstudio.github.io/r-manuals/r-ints/Tools.html#index-_r_check_pkg_sizes_) and [`_R_CHECK_PKG_SIZES_THRESHOLD_`](https://rstudio.github.io/r-manuals/r-ints/Tools.html#index-_r_check_pkg_sizes_threshold_).

```
* checking Rd line widths ... NOTE
Rd file 'h_adsl_adlb_merge_using_worst_flag.Rd':
  \examples lines wider than 100 characters:
     adlb_out <- h_adsl_adlb_merge_using_worst_flag(tern_ex_adsl, tern_ex_adlb, worst_flag = c("WGRHIFL" = "Y"))

These lines will be truncated in the PDF manual.
```

This is fixable hence let's leave it.

**Review by shajoezhu — APPROVED** [review]
approve pr to unblock merging

**Comment by pawelru** [comment]
This comment is more for `r.pkg.template` than `tern` itself.

I tried to read log output to check if correctness. According to [these lines](https://github.com/insightsengineering/r.pkg.template/blob/main/.github/workflows/build-check-install.yaml#L277-L282) I was looking for prints of env var used but I cannot find it [here](https://github.com/insightsengineering/tern/actions/runs/4162397614/jobs/7201523852#step:14:35). In particular, I am looking for a line with `_R_CHECK_CRAN_INCOMING_REMOTE_=false` as specified in this PR. There are only prints of env vars set on [workflow definition level](https://github.com/insightsengineering/r.pkg.template/blob/main/.github/workflows/build-check-install.yaml#L293-L296),

**Comment by pawelru** [comment]
> Let me jump into this PR as `tern` is the closest to CRAN submission. My long term goal for this exercise is to find collection of env vars so that R CMD CHECK has zero notes and then we can fail our check-GHA on error, warning **and** notes. Currently there are some warning/notes that needs to be fixed so here in this PR I would like to achieve the state of leaving only _fixable_ notes.
> 
> At the current state we have the following:
> 
> ```
> * checking CRAN incoming feasibility ... NOTE
> Maintainer: ‘Joe Zhu <joe.zhu@roche.com>’
> 
> Version contains large components (0.7.10.9042)
> ```
> 
> We can't avoid that until we release to CRAN. For some of the packages that would take some time. Therefore I propose to silent this note. This consists of two parts - large version components as well as new maintainer. I looked into the [source code of R](https://github.com/wch/r-source/blob/9065779ee510b7bd8ca93d08f4dd4b6e2bd31923/src/library/tools/R/QC.R#L7179-L7183) and I have found that there might be a env var that could silent it - `_R_CHECK_CRAN_INCOMING_SKIP_LARGE_VERSION_`. However, I cannot find any(!) documentation about this so I might be wrong. For the latter part - there is a `_R_CHECK_CRAN_INCOMING_` env var. It might happen that setting this one would also disable large version check so I would suggest to check this one first.
> 
> ```
> * checking installed package size ... NOTE
>   installed size is  6.5Mb
>   sub-directories of 1Mb or more:
>     doc   4.5Mb
> ```
> 
> This note is acceptable thus I propose to silent it. There are two env vars for it: [`_R_CHECK_PKG_SIZES_`](https://rstudio.github.io/r-manuals/r-ints/Tools.html#index-_r_check_pkg_sizes_) and [`_R_CHECK_PKG_SIZES_THRESHOLD_`](https://rstudio.github.io/r-manuals/r-ints/Tools.html#index-_r_check_pkg_sizes_threshold_).
> 
> ```
> * checking Rd line widths ... NOTE
> Rd file 'h_adsl_adlb_merge_using_worst_flag.Rd':
>   \examples lines wider than 100 characters:
>      adlb_out <- h_adsl_adlb_merge_using_worst_flag(tern_ex_adsl, tern_ex_adlb, worst_flag = c("WGRHIFL" = "Y"))
> 
> These lines will be truncated in the PDF manual.
> ```
> 
> This is fixable hence let's leave it.

I am sorry. I tend to confuse this issue scope with the one we have already discussed some time ago - https://github.com/insightsengineering/idr-tasks/issues/266. The blocklist mentioned there is the way how to implement fails on notes. Let's stick to the initial scope of the issue which is enabling `--as-cran` alongside with env var removing one check that resulted in warning. This has been correctly done. I will create a separate issue to populate note blocklist.

**Review by pawelru — APPROVED** [review]

**Pull request merged**